pkg/runtime.v2/framework/plugins/jobset/jobset.go

```diff
@@ -126,6 +127,24 @@ func jobSetIsSuspended(jobSet *jobsetv1alpha2.JobSet) bool {
 	return ptr.Deref(jobSet.Spec.Suspend, false)
 }
 
+func (j *JobSet) TerminalCondition(ctx context.Context, trainJob *kubeflowv2.TrainJob) (*metav1.Condition, error) {
+	jobSet := &jobsetv1alpha2.JobSet{}
+	if err := j.client.Get(ctx, client.ObjectKeyFromObject(trainJob), jobSet); err != nil {
+		return nil, err
+	}
+	if completed := meta.FindStatusCondition(jobSet.Status.Conditions, string(jobsetv1alpha2.JobSetCompleted)); completed != nil && completed.Status == metav1.ConditionTrue {
+		completed.Reason = fmt.Sprintf("%sDueTo%s", completed.Type, completed.Reason)
```

Comment:
The Reason can't include spaces ?
E.g. here we say that `FailedDueTo...`
Should we use Message for more detailed explantation ?
For example:
```go
Type: Failed
Reason: JobSetReachedMaxRestarts
Message: jobset failed due to reaching max number of restarts <- Message from JobSet
```

Reply:
The reason is machine-readable field, and should be CamelCase.

Additionally, As you can see in the following, we already set the JobSet message: https://github.com/kubeflow/training-operator/blob/d5230f7157edc7a75a58361387519ebda9a190ee/test/integration/controller.v2/trainjob_controller_test.go#L426

Reply:
That makes sense. Why you don't want to just transition reason from failed/completed JobSet to the TrainJob reason ?

Reply:
I was wondering if other arbitrary Job CRDs have less context Reason like just "Finished".
So, I connected Type and Reason by the `DueTo` so that we can use the same form Reason in all Job CRDs.
However, the TerminalCondition will be implemented by the external developer, and then they can return arbitrary conditions.

As reconsidering this, we can just directly use the JobSet Reason without connecting with Type since we do not have any guarantees at an interface level that the plugin provides a uniform format.

Reply:
I revised this one.